craft_providers/lxd/launcher.py

```diff
@@ -117,6 +119,81 @@ def _ensure_project_exists(
         )
 
 
+def _formulate_base_instance_name(
+    *, image_name: str, image_remote: str, compatibility_tag: str
+) -> str:
+    """Compute the base instance name.
+
+    :param image_remote: Name of source image's remote (e.g. ubuntu).
+    :param image_name: Name of source image (e.g. 20.04). The image name should include
+    the architecture to ensure uniqueness amongst multiple architectures built on the
+    same platform.
+    :param compatibility_tag: Compatibility tag of base configuration applied to the
+    base instance.
+
+    :returns: Name of (compatible) base instance.
+    """
+    return "-".join(["base-instance", compatibility_tag, image_remote, image_name])
+
+
+# TODO: Implement this stubbed function (CRAFT-1341)
+# pylint: disable-next=unused-argument
+def _is_valid(instance: LXDInstance) -> bool:
+    """Check if a instance is valid (i.e. the base instance is not expired).
+
+    :returns: True if the instance is valid.
+    """
+    return True
+
+
+def _launch_existing_instance(
+    *,
+    instance: LXDInstance,
+    auto_clean: bool,
+    base_configuration: Base,
+    ephemeral: bool,
+) -> bool:
+    """Start and warmup an existing instance.
+
+    :param instance: LXD instance to launch
+    :param auto_clean: If true, clean incompatible instances.
+    :param base_configuration: Base configuration to apply to the instance.
+    :param ephemeral: If the instance already exists, delete it. After the instance
```

Comment:
> the function will delete the instance if it exists, re-create it (somehow) and then later delete it after it stops

This is how I implemented it! This is a small behavioral change I grabbed from an outstanding TODO.

Before
```
# TODO: warn (or auto clean) if ephemeral
```

After:
```
    if ephemeral:
        logger.warning("Instance exists and is ephemeral. Cleaning instance.")
        instance.delete()
        return False
```

The problem is that we can't convert a non-ephemeral instance to an ephemeral instance, so I chose to delete the instance instead of just logging a warning.

We could probably support this, but making an non-ephemeral instance ephemeral is not a normal use-case. 

I reworded the docstring to explain it better.

Reply:
Thanks for the explanation. I guess I must still be missing something because the `return False` after the `delete()` means that the function is done, and the *caller* must re-create the instance and re-launch it, right? But the way the docstring was worded made me think that the function itself would do it.